pyuvdata/utils.py

```diff
@@ -1496,16 +1512,24 @@ def quadmean_collapse(arr, weights=None, axis=None, return_weights=False):
         Axis or axes to collapse (passed to np.sum). Default is all.
     return_weights : bool
         Whether to return sum of weights.
-
+    return_weights_square: bool
+        whether to return the sum of the squares of the weights. Default is False.
     """
-    out = mean_collapse(np.abs(arr)**2, weights=weights, axis=axis, return_weights=return_weights)
-    if return_weights:
+    out = mean_collapse(np.abs(arr)**2, weights=weights, axis=axis,
+                        return_weights=return_weights,
+                        return_weights_square=return_weights_square)
+    if return_weights and return_weights_square:
+        return np.sqrt(out[0]), out[1], out[2]
+    elif return_weights or return_weights_square:
         return np.sqrt(out[0]), out[1]
+    elif return_weights_square:
+        return np.sqrt(out[0]), out[2]
```

Comment:
It looks like this line isn't currently being covered by tests.

Reply:
Thanks! It looks like I managed to fix this.